src/main/java/org/gitlab4j/api/GroupApi.java

```diff
@@ -1640,11 +1676,29 @@ public Badge addBadge(Object groupIdOrPath, String linkUrl, String imageUrl) thr
      * @throws GitLabApiException if any exception occurs
      */
     public Badge editBadge(Object groupIdOrPath, Long badgeId, String linkUrl, String imageUrl) throws GitLabApiException {
-	GitLabApiForm formData = new GitLabApiForm()
-		.withParam("link_url", linkUrl, false)
-		.withParam("image_url", imageUrl, false);
-	Response response = putWithFormData(Response.Status.OK, formData, "groups", getGroupIdOrPath(groupIdOrPath), "badges", badgeId);
-	return (response.readEntity(Badge.class));
+    return (editBadge(groupIdOrPath, badgeId, null, linkUrl, imageUrl));
+    }
+
+    /**
+     * Edit a badge of a group.
+     *
+     * <pre><code>GitLab Endpoint: PUT /groups/:id/badges</code></pre>
+     *
+     * @param groupIdOrPath the group ID, path of the group, or a Group instance holding the group ID or path
+     * @param badgeId the ID of the badge to edit
+     * @param name The name of the badge to edit (may be null)
+     * @param linkUrl the URL of the badge link
+     * @param imageUrl the URL of the image link
+     * @return a Badge instance for the edited badge
+     * @throws GitLabApiException if any exception occurs
+     */
+    public Badge editBadge(Object groupIdOrPath, Long badgeId, String name, String linkUrl, String imageUrl) throws GitLabApiException {
```

Comment:
I check on the current documentation of [group badges](https://docs.gitlab.com/ee/api/group_badges.html) and I don't find the edit with name. 
I did not try against latest versions of GitLab. Did you tried?

Reply:
Yes, you cannot directly edit based on name and that is not what this code is doing.
It is still using Id to find the badge, and then setting the name and other values to that requested.
The name only really matters on creation.

Reply:
Ah, I see what you mean now.
I will go and check.

Reply:
Testing in Gitlab shows that `name` does form part of the group badge `put` request, but this is not reflected in their documentation.
I will also log a bug with them about that, must be just an oversight as it is documented for projects.

![edit_group_badge](https://user-images.githubusercontent.com/38717523/213537502-f5ad4cd5-b327-4261-b68c-889071e25e30.png)

Reply:
I have submitted a [merge request](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/109473) to update the API documentation.

Reply:
[109473](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/109473) has now been merged and you can see the updated `name` information in the [Group](https://docs.gitlab.com/ee/api/group_badges.html#edit-a-badge-of-a-group) and Project](https://docs.gitlab.com/ee/api/project_badges.html#edit-a-badge-of-a-project) API docs.

Reply:
@roadSurfer thanks.